File: app/src/main/java/com/example/yeshasprabhakar/todo/TodoViewModel.java
package com.example.yeshasprabhakar.todo;

import android.app.Application;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import java.util.List;

public class TodoViewModel extends AndroidViewModel {

    private TodoRepository mRepository;
    private LiveData<List<TodoItem>> mAllTodos;

    public TodoViewModel(Application application) {
        super(application);
        mRepository = new TodoRepository(application);
        mAllTodos = mRepository.getAllTodos();
    }

    public LiveData<List<TodoItem>> getAllTodos() {
        return mAllTodos;
    }

    public void insert(TodoItem todoItem) {
        mRepository.insert(todoItem);
    }

    public void deleteByNameDateTime(String name, String date, String time) {
        mRepository.deleteByNameDateTime(name, date, time);
    }
}

File: app/src/main/java/com/example/yeshasprabhakar/todo/TodoViewModel.kt
package com.example.yeshasprabhakar.todo

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData

class TodoViewModel(application: Application) : AndroidViewModel(application) {

    private val repository: TodoRepository // Renamed for Kotlin style
    val allTodos: LiveData<List<TodoItem>> // Renamed and made public val

    init {
        repository = TodoRepository(application)
        allTodos = repository.allTodos // Accessing the property directly from Kotlin TodoRepository
    }

    // The getAllTodos() method is removed as allTodos is now a public val.

    fun insert(todoItem: TodoItem) {
        repository.insert(todoItem)
    }

    fun deleteByNameDateTime(name: String, date: String, time: String) {
        repository.deleteByNameDateTime(name, date, time)
    }
}
